src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -237,11 +237,12 @@ const char* ClassPathEntry::copy_path(const char* path) {
   return copy;
 }
 
-ClassFileStream* ClassPathDirEntry::open_stream(const char* name, TRAPS) {
+ClassFileStream* ClassPathDirEntry::open_stream(const char* name) {
   // construct full path name
   assert((_dir != NULL) && (name != NULL), "sanity");
   size_t path_len = strlen(_dir) + strlen(name) + strlen(os::file_separator()) + 1;
-  char* path = NEW_RESOURCE_ARRAY_IN_THREAD(THREAD, char, path_len);
+  Thread* thread = Thread::current();
+  char* path = NEW_RESOURCE_ARRAY_IN_THREAD(thread, char, path_len);
```

Comment:
No need to introduce a local when only one use.

Reply:
Fixed.